src/test.c

```diff
@@ -452,9 +402,13 @@ void test_hash_avx_4_6_blocks() {
 
 void test_hash_avx_8() {
     uint32_t a = 0, b = 0, c = 0, d = 0;
-    __get_cpuid_count(7,0,&a,&b,&c,&d);
+    __get_cpuid_count(7, 0, &a, &b, &c, &d);
     if (b & bit_AVX2) {
         unsigned char digest[512];
+        uintptr_t stack_ptr;
+        asm volatile("mov %%rsp, %0" : "=r"(stack_ptr));
+        int is_stack_aligned = (stack_ptr % 16) == 0;  // Windows requires 16-byte alignment before call
+        TEST_CHECK(is_stack_aligned);
```

Comment:
Simply adding this check makes the test pass on Windows with `-O3`, this needs further investigation.